test/framework/src/main/java/org/elasticsearch/test/rest/RestActionTestCase.java

```diff
@@ -66,7 +66,10 @@ protected RestController controller() {
     protected void dispatchRequest(RestRequest request) {
         FakeRestChannel channel = new FakeRestChannel(request, false, 1);
         ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
-        controller.dispatchRequest(request, channel, threadContext);
```

Comment:
this in fact was not really chaning the context used in `RestController` - not in all places.
In same places the thread context fetched from thread pool was used